core/src/main/java/com/linecorp/armeria/common/util/Exceptions.java

```diff
@@ -167,9 +171,17 @@ public static boolean isExpected(Throwable cause) {
                 return true;
             }
 
-            if (cause instanceof Http2Exception && IGNORABLE_HTTP2_ERROR_MESSAGE.matcher(msg).find()) {
-                // Can happen when disconnected prematurely.
-                return true;
+            if (cause instanceof Http2Exception) {
+                if (IGNORABLE_HTTP2_ERROR_MESSAGE_STREAM_CLOSED.matcher(msg).find()) {
```

Comment:
Question: Do we need to check if the error is `PROTOCOL_ERROR`?
Can we combine two patterns?

Reply:
> Question: Do we need to check if the error is PROTOCOL_ERROR?

Let me check how the error is raised.

>  Can we combine two patterns?

The two pattern is `IGNORABLE_HTTP2_ERROR_MESSAGE_STREAM_CLOSED` and `IGNORABLE_HTTP2_ERROR_MESSAGE_GOAWAY`?
In my opinion, it would be better to keep separate patterns because:
- As the exceptions will not be raised frequently, the regular expression cost would not be expensive. 
- It is difficult to manage if we add a new pattern.

Reply:
> > Question: Do we need to check if the error is PROTOCOL_ERROR?
> 
> Let me check how the error is raised.
> 

The error seems to be raised with `STREAM_CLOSED`.
https://github.com/netty/netty/blob/e5951d46fc89db507ba7d2968d2ede26378f0b04/codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java#L105
https://github.com/netty/netty/blob/e5951d46fc89db507ba7d2968d2ede26378f0b04/codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java#L481-L482